packages/plugin-page-view-tracking/src/page-view-tracking.ts

```diff
@@ -0,0 +1,75 @@
+import { IAmplitudeBrowser } from '@amplitude/analytics-browser';
+import {
+  BaseEvent,
+  BeforePlugin,
+  BrowserClient,
+  BrowserConfig,
+  Campaign,
+  Event,
+  PluginType,
+  Storage,
+} from '@amplitude/analytics-types';
+import { PluginCampaignTracker } from './plugin-campaign-tracker';
+import { PageTrackingBrowserOptions } from './typings/page-view-tracking';
+
+export class PageViewTrackingPlugin implements BeforePlugin {
+  name = 'page-view-tracking';
+  type = PluginType.BEFORE as const;
+
+  constructor(
+    private readonly instance: BrowserClient | IAmplitudeBrowser,
```

Comment:
Nice way to avoid the event bus for now. I think this approach works for this project. In the future though this would prevent us from adding the Plugin during init().

Reply:
Agree. Initially I was thinking of adding instance as a parameter when we call `Plugin.setup` function, so that we did not need to pass the instance. But it would make inconsistency with other core SDKs.